src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeIngredients.java

```diff
@@ -0,0 +1,399 @@
+		"\tset the recipe addition item to netherite ingot named \"Pure Netherite\"",
+		"\tset the recipe result to netherite helmet named \"Pure Helmet\""
+})
+public class ExprRecipeIngredients extends PropertyExpression<Recipe, ItemStack> {
+
+	enum RecipePattern {
+		INGREDIENTS("recipe ingredients", "recipe ingredients", CraftingRecipeEvent.class, "This can only be used when registering a Shaped or Shapeless Recipe."),
+		FIRSTROW("recipe ingredients of [the] (1st|first) row", "recipe ingredients of the first row", ShapedRecipeEvent.class,
+			"This can only be used when registering a Shaped Recipe."),
+		SECONDROW("recipe ingredients of [the] (2nd|second) row", "recipe ingredients of the first row", ShapedRecipeEvent.class,
+			"This can only be used when registering a Shaped Recipe."),
+		THIRDROW("recipe ingredients of [the] (3rd|third) row", "recipe ingredients of the first row", ShapedRecipeEvent.class,
+			"This can only be used when registering a Shaped Recipe."),
+		INPUT("recipe (input|source) [item]", "recipe input item", new Class[]{CookingRecipeEvent.class, StonecuttingRecipeEvent.class},
+			"This can only be used when registering a Cooking, Blasting, Furnace, Campfire, Smoking or Stonecutting Recipe."),
+		BASE("[recipe] base item['s]", "recipe base item's", SmithingRecipeEvent.class,
+			"This can only be used when registering a Smithing, Smithing Transform, or Smithing Trim Recipe."),
+		TEMPLATE("[recipe] template item['s]", "recipe template item's", new Class[]{SmithingTransformRecipeEvent.class, SmithingTrimRecipeEvent.class},
+			"This can only be used when registering a Smithing Transform or Smithing Trim Recipe."),
+		ADDITION("[recipe] addition[al] item['s]", "recipe additional item's", SmithingRecipeEvent.class,
+			"This can only be used when registering a Smithing, Smithing Transform or Smithing Trim Recipe.");
+
+
+		private String pattern, toString, error;
+		private Class<? extends Event> eventClass;
+		private Class<? extends Event>[] eventClasses;
+
+		RecipePattern(String pattern, String toString, Class<? extends Event> eventClass, String error) {
+			this.pattern = "[the] " + pattern + " [of %recipes%]";
+			this.toString = toString;
+			this.eventClass = eventClass;
+			this.error = error;
+		}
+
+		RecipePattern(String pattern, String toString, Class<? extends Event>[] eventClasses, String error) {
+			this.pattern = "[the] " + pattern + " [of %recipes%]";
+			this.toString = toString;
+			this.eventClasses = eventClasses;
+			this.error = error;
+		}
+	}
+
+	private static final RecipePattern[] recipePatterns = RecipePattern.values();
+
+	static {
+		String[] patterns = new String[recipePatterns.length];
+		for (RecipePattern pattern : recipePatterns) {
+			patterns[pattern.ordinal()] = pattern.pattern;
+		}
+		Skript.registerExpression(ExprRecipeIngredients.class, ItemStack.class, ExpressionType.PROPERTY, patterns);
+	}
+
+	private boolean isEvent = false;
+	private RecipePattern selectedChoice;
```

Comment:
```suggestion
	private RecipePattern selectedPattern;
```
selected choice is kind of just saying the same thing twice